### PR TITLE
Add the Opencaching GPX extension (dont't merge yet)

### PR DESCRIPTION
**Pull request by following5**
This is a preliminary implementation of the new [Opencaching GPX extension](https://github.com/opencaching/gpx-extension-v1). It is just for review / testing / discussion, so **please don't merge**! Final code should be ready in 1-2 weeks.

This adds an `<oc:cache>` section for each geoache in the GPX files. Any apps which do not know that will ignore it; it is fully backward compatible.

The `<oc:cache>` section contains OC-specific information which otherwise is missing in GPX or not included in a safely machine-readable form:

* OC cache types
* log password requirement
* waypoint codes of other platforms, e.g. GC code
* estimated time and distance for searching
* UUIDs of log entries (this links GPX log entries to Okapi)

More information like OC attributes and OC waypoint types will be added in the future. It takes some time to define them properly.

Two detail notes:

* `<oc:size>` looks redundant to `<groundspeak:container>`. However, this is only the case for native OCPL GPX. Okapi and OCDE convert the sizes "Very large" and "Nano" to Groundspeak-compatible sizes "Large" and Micro". (Putting "Very large" and "Nano" into `<groundspeak:container>` has the drawback that some applications will not recognize it, then the cache as no size at all.) To have the same OC-GPX output on all plattforms, `<oc:size>` is redundantly added for OCPL.

* Similar for `<oc:site_team_entry>`. It is redundant to the log type "OC Team Comment". However, Okapi and OCDE don't have this log type, but a "Team comment" flag which can be set for any type of log entry. Again, to have the same OC-GPX output on all plattforms, `<oc:site_team_entry>` is redundantly added for OCPL.

### CODE REVIEW

File: lib/format.gpx.inc.php [hunk 1]
```diff
@@ -129,7 +149,6 @@
 
 // These strings must not be translated
 // Ref: see http://forum.opencaching.pl/viewtopic.php?p=121737#p121737
-$gpxContainer[0] = 'Unknown';       // OC: UNDEFINED
```
Comment (following5) [inline lib/format.gpx.inc.php]: Afaik 'Unknown' size is not definied for Groundspeak GPX. All other OC GPX implementations fall back to 'Other' size here, and I think this is the best solution.

(Practically, this fallback will only be used if someone forgot to add a new OC cache size to format.gpx.inc.php. So it's not that important.)

**Comment by following5** [comment]
What is the prefered method at OCPL to resolve conflicts in open PRs? Rebase or merge-master?

**Comment by kojoty** [comment]
> What is the prefered method at OCPL to resolve conflicts in open PRs? Rebase or merge-master?

It's up to you - we have many "merge" commits and this it is the easiest way. If you want to rebase it's only better.

**Comment by following5** [comment]
Okay, then I will merge master here. But I will wait for #1006 / #1252, because the OC GPX extension somewhat increases the GPX file size.

File: lib/format.gpx.inc.php (hunk 1 above)
Comment (andrixnet) [reply lib/format.gpx.inc.php]: It existed previously, I kept it as such when I worked on file exports

**Comment by following5** [comment]
Oops, I acidentally deleted the branch. Will open a new PR later.

**Pull request closed without merging**